File: python/ds/queue.py
class queue(object):
    def __init__(self, lst = []):
        self.queue = list(lst)

    def get(self):
        if len(self.queue) == 0:
            return
        value = self.queue.pop(0)
        return value

    def put(self, value):
        self.queue.append(value)
    
    def empty(self):
        if len(self.queue) == 0:
            return True
        return False

    def size(self):
        return len(self.queue)
    
    def __str__(self):
        return self.queue.__str__()

File: c++/Linear DS/QUEUE.cpp
#include<bits/stdc++.h>
#define MAX 1000
using namespace std;

class Queue
{
	int queue[MAX];
	int front,rear;
	public:
		Queue() {
			front=rear=-1;
		}
		int enque(int e)
		{
			if(front==rear)
			{
				front=rear=0;
				queue[rear]=e;
				rear++;
			}
			else if(rear==MAX)
			{
				cout<<"Queue is full\n";
				return 0;
			}
			else{
				queue[rear]=e;
				rear++;
			}
		}
		
		int deque()
		{
			if(front>rear-1)
			{
				cout<<"Queue is Empty\n";
				return 0;
			}
			else{
				front++;
				return (queue[front-1]);
			}
		}
		
		int traverse()
		{
			for(int i=front;i<rear;i++)
			cout<<queue[i]<<" ";
		}
};

int main()
{
	int n=0,e=0;
	Queue q;
	while(1)
	{
		cout<<"1.Enque     2.Deque     3.Traverse      4.Exit\n";
		cin>>n;
		switch(n)
		{
			case 1:
				cin>>e;
				q.enque(e);
				break;
			case 2:
				q.deque();
				break;
			case 3:
				q.traverse();
				break;
			case 4:
				return 0;
			default:
				cout<<"Please choose b/w 1/2/3/4\n";
		}
	}
	
}

